### PR TITLE
Don't have backtest sells outside of a candle

### LINKED ISSUE #2277: Exit after n minutes/n candles and roi = 0
Hi there,

1/ For testing purpose of a strategy i would like to do something like (test what happens if i close my trade after 30 minutes)

minimal_roi = {'30':-1}
stop_loss = -1

image

Freqtrade is not happy with that.

2/ this type of problem appear too when using 

minimal_roi = {'0': 0.02, '30': 0}
stoploss = -0.02

image

14% of the trades have 0% roi but it is not possible because when you close the trade 30 minutes with profit > 0 to must be take into consideration.

I think in live he don't have this type of problem ?

Have a nice day.

### PR DESCRIPTION
**Pull request by xmatthias**
## Summary
Using negative ROI values should not show sells outside of the candle.

closes #2277 

## Quick changelog

- Adjust tests to allow multiple ROI values
- Adjust tests to test for this specific usecase
- Fix bug

### CODE REVIEW

**Comment by xmatthias** [comment]
Adding the don't merge label until @gaugau3000 has the chance to test this further as discussed in https://github.com/freqtrade/freqtrade/issues/2277#issuecomment-538651785.

File: tests/optimize/test_backtest_detail.py [hunk 1]
```diff
@@ -251,6 +251,21 @@
...
 
+# Test 16: Buy, hold for 65 mins, then forcesell using roi=-1
+# Causes negative profit even though sell-reason is ROI.
+# stop-loss: 10%, ROI: 10% (should not apply), -100% after 65 minutes (limits trade duration)
+tc16 = BTContainer(data=[
+    # D  O     H     L     C     V    B  S
+    [0, 5000, 5025, 4975, 4987, 6172, 1, 0],
+    [1, 5000, 5025, 4975, 4987, 6172, 0, 0],
+    [2, 4987, 5300, 4950, 5050, 6172, 0, 0],
+    [3, 4975, 5000, 4940, 4962, 6172, 0, 0],  # ForceSell on ROI (roi=-1)
```
Comment (hroff-1902) [inline tests/optimize/test_backtest_detail.py]: why does it sell on the candle No.3, if tests_ticker_interval is '1h'?

Comment (xmatthias) [reply tests/optimize/test_backtest_detail.py]: it buys on candle 1 (buy-signals are forwardshifted by one to have it aligned with real / dry mode).

because of this, candle 2 finishes after 60 minutes - and it sells 5 minutes into candle 3 (after 65 minutes).

Comment (hroff-1902) [reply tests/optimize/test_backtest_detail.py]: ah yeah, right, it buys at candle 1...

**Comment by hroff-1902** [comment]
merging since we've got a positive reply from @gaugau3000 at #2277
